addons/portal/models/portal_mixin.py

```diff
@@ -28,9 +28,8 @@ def _compute_access_url(self):
 
     def _portal_ensure_token(self):
         """ Get the current record access token """
-        if not self.access_token:
-            # we use a `write` to force the cache clearing otherwise `return self.access_token` will return False
-            self.sudo().write({'access_token': str(uuid.uuid4())})
+        if not self.sudo().access_token:
```

Comment:
I would probably remove the override here, and instead call the method as sudo. Otherwise people might find a flow involving a call to that method that somehow leads the access token. Better be sure that people call this as sudo and know what they do with security access.

Reply:
In short: hidden sudo dealing with security stuff (tokens, ...) should not contain sudo but be manually called as sudo so that the caller knows he is bypassing security.

Reply:
I generally agree with the sentiment, especially in complex flows, but in this case `_portal_ensure_token` is a private unit method whose sole purpose is to explicitly ensure the token exists, I don't see when it should ever not be called as `sudo` or when it would "hide" anything (according to its name/purpose).